channel/state/signedstate.go

```diff
@@ -92,3 +93,37 @@ func (ss SignedState) Equal(ss2 SignedState) bool {
 	}
 	return true
 }
+
+// MarshalJSON marshals the SignedState into JSON, implementing the Marshaler interface.
+func (ss SignedState) MarshalJSON() ([]byte, error) {
+
+	// Similar type but with public members:
+	type SignedState struct {
+		State State
+		Sigs  map[uint]Signature // keyed by participant index
+	}
+
+	rr := SignedState{ss.state, ss.sigs}
+
+	return json.Marshal(rr)
+
+}
+
+// UnmarshalJSON unmarshals the passed JSON into a SignedState, implementing the Unmarshaler interface.
+func (ss *SignedState) UnmarshalJSON(j []byte) error {
+
+	// Similar type but with public members:
+	type SignedState struct {
+		State State
+		Sigs  map[uint]Signature // keyed by participant index
+	}
+
+	rr := SignedState{}
+
+	err := json.Unmarshal(j, &rr)
+
+	ss.state = rr.State
+	ss.sigs = rr.Sigs
+	return err
+
+}
```

Comment:
This allows us to do `json.Marshal` on `SignedState` (and anything that embeds a `SignedState`) *despite* `SignedState` having private / unexported fields.